dll/win32/shell32/wine/shellord.c

```diff
@@ -196,11 +196,192 @@ typedef BOOL (WINAPI *GetOpenFileNameProc)(OPENFILENAMEW *ofn);
     return ret;
 }
 
+#ifdef __REACTOS__
+BOOL SHELL_GlobalCounterChanged(LONG *pCounter, SHELL_GCOUNTER_DECLAREPARAMETERS(handle, id))
+{
+    LONG count = SHELL_GlobalCounterGet(SHELL_GCOUNTER_PARAMETERS(handle, id));
+    if (*pCounter == count)
+        return FALSE;
+    *pCounter = count;
+    return TRUE;
+}
+
+EXTERN_C void SHELL32_GetDefaultShellState(SHELLSTATE *pss);
+EXTERN_C BOOL SHELL32_ReadRegShellState(void *pregshellstate);
+EXTERN_C LSTATUS SHELL32_WriteRegShellState(void *pregshellstate);
+SHELL_GCOUNTER_DEFINE_GUID(SHGCGUID_ShellState, 0x7cb834f0, 0x527b, 0x11d2, 0x9d, 0x1f, 0x00, 0x00, 0xf8, 0x05, 0xca, 0x57);
+SHELL_GCOUNTER_DEFINE_HANDLE(g_hShellState);
+#define SHELL_GCOUNTER_SHELLSTATE SHELL_GCOUNTER_PARAMETERS(g_hShellState, GLOBALCOUNTER_SHELLSETTINGSCHANGED)
+static LONG g_ShellStateCounter = 0;
+static UINT g_CachedSSF = 0;
+static struct REGSHELLSTATE { DWORD dwSize; SHELLSTATE ss; } g_ShellState;
+enum { ssf_autocheckselect = 0x00800000, ssf_iconsonly = 0x01000000,
+       ssf_showtypeoverlay = 0x02000000, ssf_showstatusbar = 0x04000000 };
```

Comment:
The lower-case here is to distinguish with the defines in the shlobj.h header that may not be visible here because these values are Vista+?

Reply:
Correct.